### PR TITLE
Graphql-powered OrderManager

### PR DESCRIPTION
**Pull request by eliotjordan**
Rebased version (hopefully complete without errors) of https://github.com/pulibrary/figgy/pull/1762. This required some extreme git surgery which necessitated recreation on a new branch. The `resolve merge conflict` commits on the original branch were particularly nasty, which is why it made sense to start over.

### CODE REVIEW

**Comment by sdellis** [comment]
@eliotjordan 🥇 Yay! You are the best. Thank you for helping with that epic rebase!

File: app/javascript/app.vue [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <template>
   <div id="order-manager">
+    <heading level="h1">The quick brown fox</heading>
```
Comment (hackmastera) [inline app/javascript/app.vue]: what does this heading do?

Comment (eliotjordan) [reply app/javascript/app.vue]: Good catch. Looks unnecessary to me.

File: .gitignore [hunk 2]
```diff
@@ -19,3 +19,4 @@
 /app/javascript/test/unit/coverage/*
 /yarn.lock
 dump.rdb
+package-lock.json
```
Comment (escowles) [inline .gitignore]: Are we ignoring `package-lock.json` now? In general, I thought it was good practice to check that in to control what versions were running in production.

File: app/javascript/helpers/apolloClient.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+const fragmentMatcher = new IntrospectionFragmentMatcher({
+  introspectionQueryResultData: {
+    __schema: {
+      types: [
+        {
+          kind: "INTERFACE",
+          name: "Resource",
+          possibleTypes: [
+            { name: "FileSet" },
+            { name: "ScannedResource" },
```
Comment (escowles) [inline app/javascript/helpers/apolloClient.js]: TODO: add the just-merged ScannedMap here too — I created https://github.com/pulibrary/figgy/issues/1777 for this.

**Comment by escowles** [comment]
This looks good to me — the only question I have is whether we want to ignore `package-lock.json` or keep checking it in.

File: .gitignore (hunk 2 above)
Comment (sdellis) [reply .gitignore]: I think that it causes more problems than it's worth. It's the source of constant conflicts that can't be manually resolved.

File: app/javascript/helpers/apolloClient.js (hunk 3 above)
Comment (sdellis) [reply app/javascript/helpers/apolloClient.js]: I can do this on this PR if you want. Need to bump the lux version anyway.

Comment (sdellis) [reply app/javascript/helpers/apolloClient.js]: Actually, never mind. I would rather fix both in a separate PR.

**Pull request merged**